### PR TITLE
Added the filter feature

### PR DESCRIPTION
**Pull request by mimranhaider2015**
# Description of change
Added filter feature in tap salesforce

### CODE REVIEW

File: tap_salesforce/salesforce/__init__.py [hunk 1]
```diff
@@ -222,6 +223,7 @@ def __init__(self,
         self.sf_client_secret = sf_client_secret
         self.session = requests.Session()
         self.access_token = None
+        self.crmObjectFilters=crmObjectFilters
```
Comment (francisace) [inline tap_salesforce/salesforce/__init__.py]: Very minor, can we make this consistent and add in space between the variables?

File: tap_salesforce/salesforce/__init__.py [hunk 2]
```diff
@@ -385,31 +389,50 @@ def get_start_date(self, state, catalog_entry):
...
+        
     def _build_query_string(self, catalog_entry, start_date, end_date=None, order_by_clause=True):
         selected_properties = self._get_selected_properties(catalog_entry)
-
-        query = "SELECT {} FROM {}".format(",".join(selected_properties), catalog_entry['stream'])
+        query = "SELECT {} FROM {} ".format(",".join(selected_properties), catalog_entry['stream'])
 
         catalog_metadata = metadata.to_map(catalog_entry['metadata'])
-        replication_key = catalog_metadata.get((), {}).get('replication-key')
+        replication_key = catalog_metadata.get((), {}).get('valid-replication-keys', [None])[0]
```
Comment (francisace) [inline tap_salesforce/salesforce/__init__.py:398]: Quick question, did the valid-replication-keys exist on all catalog_metadata?

**Review by francisace — APPROVED** [review]

File: tap_salesforce/salesforce/__init__.py (hunk 2 above)
Comment (mimranhaider2015) [reply tap_salesforce/salesforce/__init__.py:398]: Hi @francisace 
Yes, I have not found any with 'replication-key' in catalog_metadata, all have 'valid-replication-keys'

File: tap_salesforce/salesforce/__init__.py (hunk 1 above)
Comment (mimranhaider2015) [reply tap_salesforce/salesforce/__init__.py]: Sure, I will do and push again

**Pull request merged**